docs/resources/sms_task.md

```diff
@@ -0,0 +1,162 @@
+---
+subcategory: "Server Migration Service (SMS)"
+---
+
+# huaweicloud_sms_task
+
+Manages an SMS migration task resource within HuaweiCloud.
+
+## Example Usage
+
+```hcl
+variable "source_server" {}
+variable "template_id" {}
+
+resource "huaweicloud_sms_task" "migration" {
+  type             = "MIGRATE_FILE"
+  os_type          = "LINUX"
+  source_server_id = var.source_server
+  vm_template_id   = var.template_id
+  action           = "start"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the target server is located.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `type` - (Required, String, ForceNew) Specifies the type of the migration task. Available values are
+  **MIGRATE_FILE**(file-level migration) and **MIGRATE_BLOCK**(block-level migration).
+  Changing this parameter will create a new resource.
+
+  + For Linux servers, SMS supports block-level and file-level migrations. Block-level migration has
+    high efficiency but poor compatibility while file-level migration has low efficiency but excellent compatibility.
+  + For Windows servers, SMS only supports highly efficient block-level migration.
+
+* `os_type` - (Required, String, ForceNew) Specifies the OS type of the source server. The value can be **WINDOWS** and **LINUX**.
+  Changing this parameter will create a new resource.
+
+* `source_server_id` - (Required, String, ForceNew) Specifies the ID of the source server.
+  Changing this parameter will create a new resource.
+
+* `vm_template_id` - (Optional, String, ForceNew) Specifies the template used to create the target server automatically.
+   This parameter and `target_server_id` are alternative. Changing this parameter will create a new resource.
+
+* `target_server_id` - (Optional, String, ForceNew) Specifies the existing server ID as the target server.
+   This parameter and `vm_template_id` are alternative. Changing this parameter will create a new resource.
+
+* `target_server_disks` - (Optional, List, ForceNew) Specifies the disk configurations of the target server.
+  If omitted, it will be obtained from the source server. The [object](#target_server_disks_object)
+  is documented below. Changing this parameter will create a new resource.
+
+* `use_public_ip` - (Optional, Bool, ForceNew) Specifies whether to use a public IP address for migration.
+  The default value is `true`. Changing this parameter will create a new resource.
+
+* `migration_ip` - (Optional, String, ForceNew) Specifies the IP address of the target server.
+  Use the EIP of the target server if the migration network type is Internet.
+  Use the private IP address the target server if the migration network type is Direct Connect or VPN.
```

Comment:
Nits: missing of between address and the target server. The private IP address of the targer server

Reply:
fixed